packages/fx-core/tests/component/generator/generator.test.ts

```diff
@@ -443,3 +476,30 @@ describe("Generator happy path", async () => {
     mockedEnvRestore();
   });
 });
+
+describe("Generate sample using download directory", () => {
+  const tmpDir = path.join(__dirname, "tmp");
+  const sandbox = createSandbox();
+  let mockedEnvRestore = mockedEnv({});
+  const tools = new MockTools();
+  setTools(tools);
+  const ctx = createContextV3();
+  beforeEach(async () => {
+    mockedEnvRestore = mockedEnv({
+      DOWNLOAD_DIRECTORY: "true",
+    });
+  });
+
+  afterEach(async () => {
+    sandbox.restore();
+    mockedEnvRestore();
+    if (await fs.pathExists(tmpDir)) {
+      await fs.rm(tmpDir, { recursive: true });
+    }
+  });
+
+  it("generate sample using download directory", async () => {
+    const result = await Generator.generateSample(ctx, path.join(tmpDir, "test1"), "bot-sso");
```

Comment:
What if bot-sso sample is removed? Will this test case fails and break our CI on pull request?

Reply:
The axios calls are mocked now, so removing sample won't have influence.